src/herbivoresAndCarnivores.js

```diff
@@ -2,14 +2,36 @@
 
 class Animal {
   // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
   // write your code here
+  constructor(name) {
+    super(name);
+
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = !this.hidden;
+  }
 }
 
 class Carnivore extends Animal {
   // write your code here
+  bite(prey) {
+    if (prey instanceof Herbivore && !prey.hidden) {
+      prey.health -= 50;
+      Animal.alive = Animal.alive.filter((el) => el.health);
```

Comment:
It's safer to check if health is less than or equals to zero. Imagine if after the bite health would drop by 51. Than dead animals would be alive with this code.

Reply:
It makes sense, but in this task this value can be only 100, 50 or 0